src/main/webapp/app/exercises/shared/exercise-headers/exercise-headers-information/exercise-headers-information.component.ts

```diff
@@ -0,0 +1,286 @@
+        }
+    }
+
+    addSubmissionPolicyItem() {
+        if (this.submissionPolicy?.active && this.submissionPolicy?.submissionLimit) {
+            this.informationBoxItems.push(this.getSubmissionPolicyItem());
+        }
+    }
+
+    getSubmissionPolicyItem(): InformationBox {
+        return {
+            title: 'artemisApp.programmingExercise.submissionPolicy.submissionLimitTitle',
+            content: {
+                type: 'string',
+                value: `${this.numberOfSubmissions} /  ${this.submissionPolicy?.submissionLimit}`,
+            },
+            contentColor: this.submissionPolicy?.submissionLimit ? this.getSubmissionColor() : 'body-color',
+            tooltip: 'artemisApp.programmingExercise.submissionPolicy.submissionPolicyType.' + this.submissionPolicy?.type + '.tooltip',
+            tooltipParams: { points: this.submissionPolicy?.exceedingPenalty?.toString() },
+        };
+    }
+
+    getSubmissionColor() {
+        const submissionsLeft = this.submissionPolicy?.submissionLimit ? this.submissionPolicy?.submissionLimit - this.numberOfSubmissions : 2;
+        if (submissionsLeft > 1) return 'body-color';
+        else {
+            return submissionsLeft <= 0 ? 'danger' : 'warning';
+        }
+    }
+    getPointsItem(points: number | undefined, title: string): InformationBox {
+        return {
+            title: 'artemisApp.courseOverview.exerciseDetails.' + title,
+            content: {
+                type: 'string',
+                value: this.achievedPoints ? `${this.achievedPoints} / ${points}` : `0 / ${points}`,
+            },
+        };
+    }
+
+    updateSubmissionPolicyItem() {
+        this.countSubmissions();
+
+        // need to push and pop the submission policy item to update the number of submissions
+        const submissionItemIndex = this.informationBoxItems.findIndex((item) => item.title === 'artemisApp.programmingExercise.submissionPolicy.submissionLimitTitle');
+        if (submissionItemIndex !== -1) {
+            this.informationBoxItems.splice(submissionItemIndex, 1, this.getSubmissionPolicyItem());
+        }
+    }
+
+    ngOnChanges() {
+        this.course = this.course ?? getCourseFromExercise(this.exercise);
+
+        if (this.submissionPolicy?.active && this.submissionPolicy?.submissionLimit) {
+            this.updateSubmissionPolicyItem();
+        }
+        if (this.studentParticipation?.results?.length) {
+            // The updated participation by the websocket is not guaranteed to be sorted, find the newest result (highest id)
+            this.sortService.sortByProperty(this.studentParticipation.results, 'id', false);
+
+            const latestRatedResult = this.studentParticipation.results.filter((result) => result.rated).first();
+            if (latestRatedResult) {
+                this.achievedPoints = roundValueSpecifiedByCourseSettings((latestRatedResult.score! * this.exercise.maxPoints!) / 100, this.course);
+            }
+        }
+    }
```

Comment:
IMO this `ngOnChanges` method is a bit far down in the file. Can you move it up (near ngOnInit)?